objects/obj_controller/Mouse_50.gml

```diff
@@ -362,11 +362,120 @@ if (menu=16) and (cooldown<=0){
 }
 
 
+if (menu==20) and (diplomacy==10.1){
+		show_debug_message(diplomacy_pathway, diplo_text);
+		show_debug_message( diplo_text, diplo_text);
+		show_debug_message(cooldown);
+		if (diplomacy_pathway == "intro") and (cooldown <= 0){
+			if point_in_rectangle(mouse_x, mouse_y, option_selections[0].lh, option_selections[0].top, option_selections[0].rh, option_selections[0].base){
```

Comment:
can you perhaps extract a method out of this, so you won't have to repeat yourself?

something like this (but with better names):

```
function check_and_assign_demon_pathway_in_intro(index, cooldown_arg, diplo_pathway_name, force_goodbye_arg){
    if point_in_rectangle(mouse_x, mouse_y, option_selections[index].lh, option_selections[index].top, option_selections[index].rh, option_selections[index].base){
                         cooldown=cooldown_arg;
			 diplomacy_pathway =diplo_pathway_name;
			 scr_dialogue(diplomacy_pathway);
                         force_goodbye=force_goodbye_arg;
			}
}
```

Reply:
tbh i did actually try something like that but whenever i viewed the screen it just bugged out. probably just my lack of understanding of the gamemaker click events. will play around but idk